airflow/utils/file.py

```diff
@@ -208,10 +209,11 @@ def _find_path_from_directory(
 
     :return: a generator of file paths which should not be ignored.
     """
+    # A Dict of patterns, keyed using resolved, absolute paths
     patterns_by_dir: Dict[Path, List[_IgnoreRule]] = {}
 
     for root, dirs, files in os.walk(base_dir_path, followlinks=True):
-        patterns: List[_IgnoreRule] = patterns_by_dir.get(Path(root), [])
+        patterns: List[_IgnoreRule] = patterns_by_dir.get(Path(root).resolve(), [])
```

Comment:
This `resolve()` seems bad.  When `base_dir_path` itself is a symlink, this will get the wrong path. I'd expect `Path(os.path.normpath(os.path.abspath(root)))` is more what we'd want....